internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -627,6 +673,18 @@ func (m DeploymentResource) Read() sdk.ResourceFunc {
 						output.UpgradeChannel = props.AutoUpgradeProfile.UpgradeChannel
 					}
 
+					if props.NginxAppProtect != nil && props.NginxAppProtect.WebApplicationFirewallSettings.ActivationState != nil {
```

Comment:
I see that we also have `WebApplicationFirewallStatus` on the API. Is this useful information for the user and if so, we should export that out to the user as well.

Reply:
In order to expose `WebApplicationFirewallStatus`, I had to unnest the `nginx_app_protect` block. You can't have both an attribute and argument within a nested block.

Reply:
My bad @arpith-f5. I thought you did not have the computed properties exported. looks like you do that already. 👍